### PR TITLE
RavenDB-14347 updated to .NET Core 3.1

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: bench/Micro.Benchmark/Benchmarks/Hardware/Compare.cs [hunk 1]
```diff
@@ -236,7 +236,7 @@ private static int CompareScalarAltPopCount(void* p1, void* p2, int size)
             //       we can use that we know we are running on little endian hardware and the very first bit set 
             //       will correspond to the first byte which is different. 
 
-            bpx += Popcnt.PopCount((ulong)((long)xor & -(long)xor) - 1) >> 3;
+            bpx += Popcnt.PopCount((uint)((long)xor & -(long)xor) - 1) >> 3;
```
Comment (ayende) [inline bench/Micro.Benchmark/Benchmarks/Hardware/Compare.cs:239]: Why do we have all these changes?

Comment (ppekrol) [reply bench/Micro.Benchmark/Benchmarks/Hardware/Compare.cs:239]: Intrinsics is now a part of a framework and Popcnt.PopCount requires the value to be an uint.

File: src/Raven.Client/Documents/Session/Operations/StreamOperation.cs [hunk 2]
```diff
@@ -82,15 +82,15 @@ public IEnumerator<BlittableJsonReaderObject> SetResult(StreamResult response)
             return enumerator;
         }
 
-        public async Task<IAsyncEnumerator<BlittableJsonReaderObject>> SetResultAsync(StreamResult response)
+        public async Task<Raven.Client.Util.IAsyncEnumerator<BlittableJsonReaderObject>> SetResultAsync(StreamResult response)
```
Comment (ayende) [inline src/Raven.Client/Documents/Session/Operations/StreamOperation.cs:85]: Move to `System.IAsyncEnumerable`

File: src/Raven.Client/Documents/Subscriptions/DocumentSubscriptions.cs [hunk 3]
```diff
@@ -20,6 +20,7 @@
 using Raven.Client.Util;
 using Sparrow.Collections;
 using Sparrow.Json;
+using IAsyncDisposable = Raven.Client.Util.IAsyncDisposable;
```
Comment (ayende) [inline src/Raven.Client/Documents/Subscriptions/DocumentSubscriptions.cs:23]: Move to `System.IAsyncDisposable `

File: src/Raven.Server/Https/HttpsConnectionMiddleware.cs [hunk 4]
```diff
@@ -1,68 +1,43 @@
...
-
-        private async Task<IAdaptedConnection> InnerOnConnectionAsync(ConnectionAdapterContext context)
-        {
-            var sslStream = new SslStream(context.ConnectionStream,
-                leaveInnerStreamOpen: false,
-                userCertificateValidationCallback: (sender, certificate, chain, sslPolicyErrors) =>
+            options.ConfigureHttpsDefaults(o =>
+            {
+                o.SslProtocols = SslProtocols.Tls13 | SslProtocols.Tls12;
+                o.CheckCertificateRevocation = true;
```
Comment (ayende) [inline src/Raven.Server/Https/HttpsConnectionMiddleware.cs:38]: Set to false.
Check that this works in offline mode

Comment (ayende) [reply src/Raven.Server/Https/HttpsConnectionMiddleware.cs:38]: Need to check `HandshakeTimeout`

File: src/Raven.Server/ServerWide/LocalEndpointClient.cs [hunk 5]
```diff
@@ -143,7 +138,7 @@ public LocalHttpResponse(HttpContext httpContext)
                     HttpContext = httpContext;
                     Headers = new HeaderDictionary();
                     // ReSharper disable once VirtualMemberCallInConstructor
-                    Cookies = new ResponseCookies(Headers, new DefaultObjectPool<StringBuilder>(new DefaultPooledObjectPolicy<StringBuilder>()));
+                    //Cookies = new ResponseCookies(Headers, new DefaultObjectPool<StringBuilder>(new DefaultPooledObjectPolicy<StringBuilder>())); // TODO [ppekrol]
```
Comment (ayende) [inline src/Raven.Server/ServerWide/LocalEndpointClient.cs:141]: TODO here

File: src/Raven.Server/Https/HttpsConnectionMiddleware.cs (hunk 4 above)
Comment (ppekrol) [reply src/Raven.Server/Https/HttpsConnectionMiddleware.cs:38]: @iftahbe is checking that

File: src/Raven.Server/ServerWide/LocalEndpointClient.cs (hunk 5 above)
Comment (ppekrol) [reply src/Raven.Server/ServerWide/LocalEndpointClient.cs:141]: Not needed

**Pull request merged**